### PR TITLE
Fix some golint and ineffassign issues

### PR DESCRIPTION
**Pull request by thaJeztah**
Noticed a bunch of these on https://goreportcard.com/report/github.com/moby/moby

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @vdemeester

**Comment by thaJeztah** [comment]
hm.. made one mistake looks like; fixing

**Comment by thaJeztah** [comment]
Hm... guess I put an err.IsNil where it should be producing an error

```
20:02:18 FAIL: docker_cli_config_create_test.go:55: DockerSwarmSuite.TestConfigCreateResolve
20:02:18 
20:02:18 [ddf32ca613d48] waiting for daemon to start
20:02:18 [ddf32ca613d48] daemon started
20:02:18 
20:02:18 docker_cli_config_create_test.go:97:
20:02:18     c.Assert(err, checker.IsNil)
20:02:18 ... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc4205b2040), Stderr:[]uint8(nil)} ("exit status 1")
20:02:18 
20:02:18 [ddf32ca613d48] exiting daemon


20:03:20 FAIL: docker_cli_secret_create_test.go:16: DockerSwarmSuite.TestSecretCreateResolve
20:03:20 
20:03:20 [d6dac521d7c1e] waiting for daemon to start
20:03:20 [d6dac521d7c1e] daemon started
20:03:20 
20:03:20 docker_cli_secret_create_test.go:58:
20:03:20     c.Assert(err, checker.IsNil)
20:03:20 ... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc4200d3f60), Stderr:[]uint8(nil)} ("exit status 1")
20:03:20 
20:03:20 [d6dac521d7c1e] exiting daemon
```

File: integration-cli/docker_api_containers_test.go [hunk 1]
```diff
@@ -2200,7 +2201,7 @@ func (s *DockerSuite) TestContainerKillCustomStopSignal(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	defer res.Body.Close()
 
-	b, err := ioutil.ReadAll(res.Body)
+	b, _ := ioutil.ReadAll(res.Body)
```
Comment (vdemeester) [inline integration-cli/docker_api_containers_test.go]: shouldn't we check for err here too ?

File: integration-cli/docker_cli_swarm_unix_test.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ func (s *DockerSwarmSuite) TestSwarmVolumePlugin(c *check.C) {
 	defer plugin.Close()
 
 	// create a dummy volume to trigger lazy loading of the plugin
-	out, err = d.Cmd("volume", "create", "-d", "customvolumedriver", "hello")
+	d.Cmd("volume", "create", "-d", "customvolumedriver", "hello")
```
Comment (vdemeester) [inline integration-cli/docker_cli_swarm_unix_test.go]: shouldn't we check the err here ?

**Review by vdemeester — COMMENTED** [review]
SGTM :angel:

**Comment by thaJeztah** [comment]
ping @vdemeester @dnephin updated; PTAL

**Review by vdemeester — APPROVED** [review]
LGTM :frog:

**Comment by dnephin** [comment]
I wonder why these weren't flagged by hack/validate/gometalinter

**Review by boaz0 — APPROVED** [review]
LGTM :1st_place_medal:

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**